azure-pipelines.yml

```diff
@@ -0,0 +1,252 @@
+      if [[ "$INVERTED" != "ON" ]]; then
+        sudo -E apt-get -yq --no-install-suggests --no-install-recommends --allow-unauthenticated install libevent-dev libeditline-dev liblz4-dev libre2-dev protobuf-compiler libprotobuf-dev libprotoc-dev libicu-dev || exit 1;
+      fi;
+      sudo ln -s $(which ccache) /usr/lib/ccache/$CC;
+      sudo ln -s $(which ccache) /usr/lib/ccache/$CXX;
+
+      UPDATE_TIME=$SECONDS
+      echo --- Packages updated in $UPDATE_TIME seconds.
+
+      $CC -v
+      $CXX -v
+      ccache --version
+      ccache --print-config
+      ccache --zero-stats
+
+      echo '##vso[task.setvariable variable=CC]'$CC
+      echo '##vso[task.setvariable variable=CXX]'$CXX
+      echo '##vso[task.setvariable variable=UPDATE_TIME]'$UPDATE_TIME
+      echo '##vso[task.prependpath]/usr/lib/ccache'
+
+    displayName: '*** Install Build Dependencies'
+
+  - task: CacheBeta@0
+    continueOnError: true
+    inputs:
+      key: ccache | $(Agent.OS)-$(Compiler)-$(CompilerVer)-$(BuildType) | $(Build.SourceVersion)
+      restoreKeys: ccache | $(Agent.OS)-$(Compiler)-$(CompilerVer)-$(BuildType)
+      path: $(CCACHE_DIR)
+    displayName: '*** Download ccached data'
+
+  - script: |
+      echo --- Set cmake parameters
+      CMAKE_OPT="
+        -DCMAKE_BUILD_TYPE=$(BuildType)
+        -DBUILD_CONFIG=mysql_release
+        -DFEATURE_SET=community
+        -DENABLE_DOWNLOADS=1
+        -DDOWNLOAD_BOOST=1
+        -DWITH_BOOST=../deps
+        -DWITH_KEYRING_VAULT=OFF
+        -DWITH_PAM=ON
+        -DWITH_TOKUDB=ON
+        -DWITH_ROCKSDB=ON
+        -DMYSQL_MAINTAINER_MODE=OFF
+        -DALLOW_NO_SSE42=ON
+        -DWITH_CURL=system
+        -DWITH_MECAB=system
+        -DWITH_RAPIDJSON=bundled
+        -DWITH_SSL=system
+        -DWITH_LIBEVENT=bundled
+      "
+
+      if [[ "$INVERTED" != "ON" ]]; then
+        CMAKE_OPT+="
+          -DWITH_READLINE=system
+          -DWITH_ICU=system
+          -DWITH_LZ4=system
+          -DWITH_PROTOBUF=system
+          -DWITH_RE2=system
+          -DWITH_ZLIB=system
+          -DWITH_NUMA=ON
+        "
+      else
+        CMAKE_OPT+="
+          -DWITH_EDITLINE=bundled
+          -DWITH_ICU=bundled
+          -DWITH_LZ4=bundled
+          -DWITH_PROTOBUF=bundled
+          -DWITH_RE2=bundled
+          -DWITH_ZLIB=bundled
+          -DWITH_NUMA=OFF
+          -DWITH_ARCHIVE_STORAGE_ENGINE=OFF
+          -DWITH_BLACKHOLE_STORAGE_ENGINE=OFF
+          -DWITH_EXAMPLE_STORAGE_ENGINE=ON
+          -DWITH_FEDERATED_STORAGE_ENGINE=OFF
+          -DWITHOUT_PERFSCHEMA_STORAGE_ENGINE=ON
+          -DWITH_INNODB_MEMCACHED=ON
+        "
+      fi
+
+      echo --- CMAKE_OPT=\"$CMAKE_OPT\"
+      mkdir bin; cd bin
+      CC=$CC CXX=$CXX cmake .. $CMAKE_OPT || exit 1
```

Comment:
For some reason, in my experiments `cmake` never returned error return code when finished with an error :(